build_tools/github_actions/post_build_upload.py

```diff
@@ -0,0 +1,301 @@
+    else:
+        run_aws_cp(log_dir, s3_base_path, content_type="text/plain")
+
+    # Upload index.html
+    index_path = log_dir / "index.html"
+    if index_path.is_file():
+        index_s3_dest = f"{s3_base_path}/index.html"
+        run_aws_cp(index_path, index_s3_dest, content_type="text/html")
+        log(f"[INFO] Uploaded {index_path} to {index_s3_dest}")
+    else:
+        log(f"[INFO] No index.html found at {log_dir}. Skipping index upload.")
+
+
+def upload_build_summary(args):
+    external_repo_path, bucket = retrieve_bucket_info()
+    run_id = args.run_id
+    bucket_url = (
+        f"https://{bucket}.s3.amazonaws.com/{external_repo_path}{run_id}-{PLATFORM}"
+    )
+    log(f"Adding links to job summary to bucket {bucket}")
+    build_dir = args.build_dir
+    amdgpu_family = args.amdgpu_family
+
+    log_url = f"{bucket_url}/logs/{amdgpu_family}/index.html"
+    gha_append_step_summary(f"[Build Logs]({log_url})")
+    if os.path.exists(build_dir / "artifacts" / "index.html"):
+        artifact_url = f"{bucket_url}/index-{amdgpu_family}.html"
+        gha_append_step_summary(f"[Artifacts]({artifact_url})")
+    else:
+        log("No artifacts index found. Skipping artifact link.")
+
+
+def run(args):
+    log("Creating Ninja log archive")
+    log("--------------------------")
+    create_ninja_log_archive(args.build_dir)
+
+    log("Indexing log files")
+    log("------------------")
+    index_log_files(args.build_dir, args.amdgpu_family)
+
+    if args.ci or args.upload:
+        check_aws_cli_available()
+        log("Upload build artifacts")
+        log("----------------------")
+        external_repo_path, bucket = retrieve_bucket_info()
+        run_id = args.run_id
+        bucket_uri = f"s3://{bucket}/{external_repo_path}{run_id}-{PLATFORM}"
+
+        create_index_file(args)
+        upload_artifacts(args, bucket_uri)
+
+        log("Upload log")
+        log("----------")
+        upload_logs_to_s3(args.run_id, args.amdgpu_family, args.build_dir)
+
+    if args.ci and args.upload:
+        log("Upload build summary")
+        log("--------------------")
+        upload_build_summary(args)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Post Build Upload steps")
+    parser.add_argument(
+        "--build-dir",
+        type=Path,
+        default=Path(os.getenv("BUILD_DIR", "build")),
+        help="Build directory containing logs (default: 'build' or $BUILD_DIR)",
+    )
+    parser.add_argument(
+        "--amdgpu-family",
+        type=str,
+        default=os.getenv("AMDGPU_FAMILIES"),
+        required=True,
+        help="AMDGPU family name (default: $AMDGPU_FAMILIES)",
+    )
+    parser.add_argument("--run-id", type=str, help="GitHub run ID of this workflow run")
+    parser.add_argument(
+        "--ci",
+        default=False,
+        help="Enable CI steps",
+        action=argparse.BooleanOptionalAction,
+    )
```

Comment:
You could give this a default value based on the `CI` environment variable that github actions sets. See also this code in our helper library (which you could call in this file if you wanted to): https://github.com/ROCm/TheRock/blob/f89dcd5c5625baecb467b9287e952c5c819073fd/build_tools/github_actions/github_actions_utils.py#L17-L20

Then the workflows wouldn't need to explicitly pass `--ci` (I would expect scripts in `build_tools/github_actions/` to by default run in a CI environment but they can still allow running locally or in tests)